robotPositioner/robotPositioner.h

```diff
@@ -0,0 +1,382 @@
+
+        if (m_cam->captureSuperPixelGreyscale(output)) {
+            m_unwrapper.unwrap(output, unwrapped);
+            char fileName[255];
+            sprintf(fileName, "image_%u.png", m_imageNumber++);
+            cv::imwrite(fileName, unwrapped);
+            std::cout << " image " << m_imageNumber << " captured " << std::endl;
+        }
+    }
+
+//-----------------PUBLIC API---------------------------------------------------------------------
+public:
+
+    // constructor
+    robotPositioner(
+        unsigned int vicon_udp_client_port,
+        std::string  vicon_capture_control_ip_address,
+        std::string  vicon_capture_control_executable_path,
+        unsigned int capture_control_port,
+        double       robot_r,
+        double       robot_D,
+        double       allowed_heading_error,
+        double       stopping_distance,
+        std::string  video_device,
+        double       k1,
+        double       k2,
+        double       alpha,
+        double       beta,
+        double       max_velocity
+        ) : m_vicon_udp_client_port(vicon_udp_client_port),
+            m_vicon_capture_control_ip_address(vicon_capture_control_ip_address),
+            m_vicon_capture_control_executable_path(vicon_capture_control_executable_path),
+            m_capture_control_port(capture_control_port),
+            m_robot_r(robot_r),
+            m_robot_D(robot_D),
+            m_allowed_heading_error(allowed_heading_error),
+            m_stopping_distance(stopping_distance),
+            m_video_device(video_device),
+            m_k1(k1),
+            m_k2(k2),
+            m_alpha(alpha),
+            m_beta(beta),
+            m_max_velocity(max_velocity)
+        {
+
+        m_imageNumber = 0;
+        m_bot = new BoBRobotics::Robots::Norbot(); // init robot
+        setupVicon();                              // init vicon system
+
+        // setup 360 camera if exists
+        try {
+            setupCamera360();
+            std::cout << "camera 360 finished setting up" << std::endl;
+        } catch(...) {
+            std::cout << "camera cannot be initialised, maybe it does not exists..." << std::endl;
+            m_hasCamera = false;
+        }
+    }
+
+    //destructor
+    ~robotPositioner() {
+
+        if(!m_viconCaptureControl->stopRecording("test1")) {
+            std::cout << "failed to stop recording" << std::endl;
+        }
+
+        delete m_vicon;
+        delete m_viconCaptureControl;
+        delete m_bot;
+
+        if (m_hasCamera) {
+            delete m_cam;
+        }
+    }
+
+    /**
+    *	starts the robot positioning program. The function expects a csv file with 4 columns |x|y|z|a|
+    *   where 'xyz' are the coordinates of the goal location and 'a' is the final angle.
+    *   the robot will visit all the coordinates in the file, then snap an image with the camera at each 
+    *   location if the camera is attached.
+    **/
+    void startSession(const std::string fileName) {
```

Comment:
I think this functionality and the logic to read coordinates doesn't belong here - I think for now it's probably best to just move it into your example application.

Reply:
Yep.

On a related note: your example application could go into the examples/ folder and the header should live somewhere else with the other library code... maybe let's put it in common/ for now.